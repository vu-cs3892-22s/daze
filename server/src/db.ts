import { Pool } from 'node-postgres';
import dotenv from 'dotenv';
import { resolve } from 'path/posix';

dotenv.config()

const pool = new Pool({
  user: process.env.DB_USER,
  host: process.env.DB_HOST,
  database: process.env.DB_NAME,
  password: process.env.DB_PASSWORD,
  port: Number(process.env.DB_PORT),
  ssl: { rejectUnauthorized: false }
});

pool.on('error', (err, client) => {
  console.error('Unexpected error on idle client', err);
  //process.exit(-1);
})


export async function addUser(body: any){
  const text = `
    INSERT INTO users ("VanderbiltEmail", "FirstName", "LastName", "PhoneNumber")
    VALUES ('${body.VanderbiltEmail}', '${body.FirstName}', '${body.LastName}', '${body.PhoneNumber}')
    RETURNING "VanderbiltEmail"
  `;
  /*
  
  async function callQuery() {

    const client = await pool.connect()
    try {
      const res = await client.query(text)
      console.log(res.rows[0])
      resolve("0")
    } catch (err: any) {
      console.log(err.stack)
      console.log("this is the err.error_code " + err.code)
      error_code = err.error_code
      resolve(err.code)
    }
    finally {
      client.release()
    }

  }
  let value = await callQuery();
  return value; */

  pool.connect()
  .then(client => {
    return client
      .query(text)
      .then(res => {
        client.release();
        console.log(res.rows[0]);
      })
      .catch(err => {
        client.release();
        console.log(err.stack);
      })
  })  
} 

export function obtainUser(email: any){
  const text = `SELECT * FROM users WHERE VanderbiltEmail = $1`;
  const values = [email];
  
  pool.connect()
  .then(client => {
    return client
      .query(text, values)
      .then(res => {
        client.release()
        console.log(res.rows[0])
      })
      .catch(err => {
        client.release()
        console.log(err.stack)
      })
  })
}



