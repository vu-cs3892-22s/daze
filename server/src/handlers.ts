import express from 'express';

export const createUser = (req: express.Request, res: express.Response) => {
  // Insert DB function
  res.send({
    message: 'Creating new user'
  });
};

export const updateUser = (req: express.Request, res: express.Response) => {
  // Insert DB function
  res.send({
    message: 'Updating user'
  });
};

export const getUser = (req: express.Request, res: express.Response) => {
  const vunetId = req.params.vunet_id;
  // Insert DB function
  res.send({
    message: `Getting user ${vunetId}`
  });
};

export const submitData = (req: express.Request, res: express.Response) => {
  // Insert DB function
  res.send({
    message: 'Posting new line data'
  });
};

export const submitComments = (req: express.Request, res: express.Response) => {
  // Insert DB function
  res.send({
    message: 'Posting new comment'
  });
};

export const getDiningHallInfo = (
  req: express.Request,
  res: express.Response
) => {
  const diningHallName = req.params.dininghall_name;
  // Insert DB function
  res.send({
    message: `Getting info for ${diningHallName}`
  });
};
