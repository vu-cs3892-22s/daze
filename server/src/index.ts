export const func = (x: any) => {
  console.log(x);
};

export const func2 = (x: any) => {
  return x + 2;
};
