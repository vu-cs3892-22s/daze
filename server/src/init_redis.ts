import * as redis from 'redis';
import dotenv from 'dotenv';

dotenv.config({
  path: `${__dirname}/../../.env`
});

// export const client = redis.createClient({
//   url: process.env.REDIS_TLS_URL,
//   socket: {
//     tls: true,
//     rejectUnauthorized: false
//   }
// });
export const client = redis.createClient();

(async () => {
  try {
    await client.connect();
  } catch (e) {
    console.log('Redis failed to connect');
  }
})();

client.on('connect', () => {
  console.log('Client connected to redis');
});

client.on('ready', () => {
  console.log('Client connected to redis and ready to use.');
});

client.on('error', (err: any) => {
  console.log(err);
});

client.on('end', () => {
  console.log('Client disconnected from redis');
});

process.on('SIGINT', () => {
  client.quit();
});
