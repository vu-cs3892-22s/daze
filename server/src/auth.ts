import express from 'express';
import passport from 'passport';
import { Strategy } from 'passport-google-oauth2';
import { queryCreateUser, queryUpdateUser, queryGetUser } from './db';

type User = {
  email?: string;
  firstName?: string;
  lastName?: string;
  phoneNumber?: string;
};

passport.serializeUser((user, done) => {
  console.log('serializing user:', user);
  return done(null, user);
});

passport.deserializeUser(async (user: User, done) => {
  try {
    console.log('user:', user);
    done(null, user);
  } catch (err) {
    done(err);
  }
});

const googleStrategy = new Strategy(
  {
    clientID: process.env.GOOGLE_CLIENT_ID as string,
    clientSecret: process.env.GOOGLE_CLIENT_SECRET as string,
    callbackURL: 'http://localhost:8080/auth/google/callback',
    scope: ['profile', 'email'],
    // TODO: fix this ^^
    passReqToCallback: true
  },
  // TODO: fix types
  async function (
    request: unknown,
    accessToken: unknown,
    refreshToken: unknown,
    profile: any,
    done: any
  ) {
    // TODO: check what's actually in the profile
    try {
      let user = await queryGetUser(profile.emails[0].value);
      if (user === undefined) {
        // User not found, create user
        user = await queryCreateUser({
          vanderbiltEmail: profile.emails[0].value,
          firstName: profile.name.givenName,
          lastName: profile.name.familyName,
          phoneNumber: ''
        });
      }
      console.log('hey user:', user);
      done(null, user);
    } catch (err) {
      console.log('yo this failed');
      done(err);
    }
  }
);

passport.use(googleStrategy);

console.log('hey:', process.env.GOOGLE_CLIENT_ID);

export { passport };

// Auth routes
// const logout = (req: express.Request, res: express.Response) => {
//     req.logout();
//     res.redirect('/');
//   }

//   const

//   app.get('/auth/linkedin', passport.authenticate('linkedin', (err, user, info) => {
//     if (err) { return next(err) }
//   }), async (req, res, next) => {
//     // The request will be redirected to LinkedIn for authentication, so this
//     // function will not be called.
//     console.log('auth/linkedin',req)
//     res.redirect('/')
//     next()
//   })

//   app.get('/auth/linkedin/callback', passport.authenticate('linkedin', {
//       successRedirect: '/auth/linkedin/redirect',
//       failureRedirect: '/auth/linkedin'
//     }), async (req, res, next) => {
//         // console.log(req.user.dataValues)
//         res.send(req.user.dataValues)
//   })

//   // Redirect the user back to the app
//   app.get('/auth/linkedin/redirect', async (req, res, next) => {
//   // you can see what you get back from LinkedIn here:
//   console.log(req.user.dataValues)
//   res.redirect(<deep-link-to-react-native-app>)
//   })
