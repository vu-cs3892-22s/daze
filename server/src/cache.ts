import { client } from './init_redis';

interface DataBody {
  vanderbiltEmail: string;
  diningHallName: string;
  lineLength: string;
  timeStamp: string;
}

const diningHallNames = [
  '2301',
  'Commons',
  'EBI',
  'Kissam',
  'McTyeire',
  'Rand_Bowls',
  'Rand_Randwich',
  'Rand_Fresh_Mex',
  'Rand_Mongolian',
  'Rand_Chicken_Shack',
  'Zeppos',
  'Alumni',
  'Grins',
  'Holy_Smokes',
  'Local_Java',
  'Suzies_Blair',
  'Suzies_FGH',
  'Suzies_MRB',
  'Food_For_Thought'
];

const diningHallLocations: { [key: string]: number[] } = {
  2301: [36.1465882, -86.8035787],
  Commons: [36.1418863, -86.7971635],
  EBI: [36.1486501, -86.8037112],
  Kissam: [36.1493254, -86.8018191],
  McTyeire: [36.1437784, -86.8032918],
  Rand_Bowls: [36.1464751, -86.8033662],
  Rand_Randwich: [36.1464751, -86.8033662],
  Rand_Fresh_Mex: [36.1464751, -86.8033662],
  Rand_Mongolian: [36.1464751, -86.8033662],
  Rand_Chicken_Shack: [36.1464751, -86.8033662],
  Zeppos: [36.1468383, -86.807612],
  Alumni: [36.1479339, -86.803365],
  Grins: [36.144731, -86.806367],
  Holy_Smokes: [36.144634, -86.8040498],
  Local_Java: [36.1462604, -86.8037347],
  Suzies_Blair: [36.138448, -86.805293],
  Suzies_FGH: [36.1445184, -86.8036674],
  Suzies_MRB: [36.1445865, -86.8005806],
  Food_For_Thought: [36.1453047, -86.8008198]
};

const diningHallTypes: { [key: string]: string } = {
  2301: "Residential Dining Hall",
  Commons: "Residential Dining Hall",
  EBI: "Residential Dining Hall",
  Kissam: "Residential Dining Hall",
  McTyeire: "Residential Dining Hall",
  Rand_Bowls: "Residential Dining Hall",
  Rand_Randwich: "Residential Dining Hall",
  Rand_Fresh_Mex: "Residential Dining Hall",
  Rand_Mongolian: "Residential Dining Hall",
  Rand_Chicken_Shack: "Residential Dining Hall",
  Zeppos: "Residential Dining Hall",
  Alumni: "Cafe",
  Grins: "Cafe",
  Holy_Smokes: "Cafe",
  Local_Java: "Cafe",
  Suzies_Blair: "Cafe",
  Suzies_FGH: "Cafe",
  Suzies_MRB: "Cafe",
  Food_For_Thought: "Cafe",
};

export const insertData = async (data: DataBody, db: number) => {
  await client.select(db);
  const diningHallName = data.diningHallName;
  await client.rPush(diningHallName, JSON.stringify(data));
};

export const getDataForDiningHall = async (
  diningHallName: string,
  db: number
) => {
  await client.select(db);
  const result = await client.lRange(diningHallName, 0, -1);
  return result;
};

export const getDataForDiningHalls = async (db: number) => {
  const results: any = {};
  await client.select(db);
  for (const diningHallName of diningHallNames) {
    const result = await client.lRange(diningHallName, 0, -1);
    const long = diningHallLocations[diningHallName][0];
    const lat = diningHallLocations[diningHallName][1];
    const type = diningHallTypes[diningHallName];
    const name = diningHallName;
    results[diningHallName] = {
      lineLength: result,
      longitude: long,
      latitude: lat,
      name: name,
      type: type
    };
  }

  return results;
};
