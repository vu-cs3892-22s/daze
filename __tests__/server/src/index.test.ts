import * as index from '../../../server/src/index';
// @ponicode
describe('index.func', () => {
  test('0', () => {
    const result: any = index.func('foo bar');
    expect(result).toMatchSnapshot();
  });

  test('1', () => {
    const result: any = index.func('Foo bar');
    expect(result).toMatchSnapshot();
  });

  test('2', () => {
    const result: any = index.func('Hello, world!');
    expect(result).toMatchSnapshot();
  });

  test('3', () => {
    const result: any = index.func('');
    expect(result).toMatchSnapshot();
  });
});
