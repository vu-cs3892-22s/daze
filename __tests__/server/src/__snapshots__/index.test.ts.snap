// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.func 0 1`] = `undefined`;

exports[`index.func 1 1`] = `undefined`;

exports[`index.func 2 1`] = `undefined`;

exports[`index.func 3 1`] = `undefined`;
